cmd/statusd/main.go

```diff
@@ -106,17 +124,27 @@ func makeNodeConfig() (*params.NodeConfig, error) {
 		nodeConfig.LogFile = *logFile
 	}
 
-	nodeConfig.LightEthConfig.Enabled = true
 	nodeConfig.RPCEnabled = *httpEnabled
 	nodeConfig.WhisperConfig.Enabled = *whisperEnabled
-	nodeConfig.SwarmConfig.Enabled = *swarmEnabled
+
+	nodeConfig.HTTPPort = *httpPort
+	nodeConfig.IPCEnabled = *ipcEnabled
+
+	if *whisperEnabled {
+		// disable les and swarm for wnode
+		nodeConfig.LightEthConfig.Enabled = false
+		nodeConfig.SwarmConfig.Enabled = false
+
+		return whisperConfig(nodeConfig)
+	}
 
 	// RPC configuration
 	if !*httpEnabled {
 		nodeConfig.HTTPHost = "" // HTTP RPC is disabled
 	}
-	nodeConfig.HTTPPort = *httpPort
-	nodeConfig.IPCEnabled = *ipcEnabled
+
+	nodeConfig.LightEthConfig.Enabled = true
```

Comment:
adjusting so that only setter  is:
nodeConfig.SwarmConfig.Enabled = *swarmEnabled